unset doc is updated by invalid value of $inc operator but it shouldn't

### Versions

0.7.1

### What did you do?

Moved $inc to compat test and test $inc. The compat test tests $inc on unset.

### What did you expect to see?

The invalid `string`  value is updated by `$inc` when unset data is used. It should return error like compat.

```
        --- FAIL: TestUpdateFieldCompatInc/IncWithStringValue/TestUpdateFieldCompatInc-IncWithStringValue_Unsets (0.03s)
            --- FAIL: TestUpdateFieldCompatInc/IncWithStringValue/TestUpdateFieldCompatInc-IncWithStringValue_Unsets/unset (0.02s)
                update_field_compat_test.go:141: 
                        Error Trace:    /home/chi/go/src/github.com/chilagrow/FerretDB/integration/update_compat_test.go:116
                        Error:          Received unexpected error:
                                        write exception: write errors: [Cannot increment with non-numeric argument: {v: "bad value"}]
                        Test:           TestUpdateFieldCompatInc/IncWithStringValue/TestUpdateFieldCompatInc-IncWithStringValue_Unsets/unset
                        Messages:       compat error; target returned no error
```

### What did you see instead?

Same error as compat

--- a/integration/update_field_compat_test.go
+++ b/integration/update_field_compat_test.go
@@ -210,12 +210,16 @@ func TestUpdateFieldCompatIncComplex(t *testing.T) {
 			skip:   "https://github.com/FerretDB/FerretDB/issues/1658",
 		},
 		"IncOnString": {
-			update: bson.D{{"$inc", "string"}},
-			skip:   "https://github.com/FerretDB/FerretDB/issues/1660",
+			update:     bson.D{{"$inc", "string"}},
+			resultType: emptyResult,
 		},
 		"IncWithStringValue": {
-			update: bson.D{{"$inc", bson.D{{"v", "bad value"}}}},
-			skip:   "https://github.com/FerretDB/FerretDB/issues/1660",
+			update:     bson.D{{"$inc", bson.D{{"v", "bad value"}}}},
+			resultType: emptyResult,
+		},
+		"NotExistStringValue": {
+			update:     bson.D{{"$inc", bson.D{{"foo.bar", "bad value"}}}},
+			resultType: emptyResult,
 		},
 	}
 

--- a/internal/handlers/common/update.go
+++ b/internal/handlers/common/update.go
@@ -250,6 +250,17 @@ func processIncFieldExpression(doc *types.Document, updateV any) (bool, error) {
 		path := types.NewPathFromString(incKey)
 
 		if !doc.HasByPath(path) {
+			// ensure incValue is a valid number type.
+			switch incValue.(type) {
+			case float64, int32, int64:
+			default:
+				return false, NewWriteErrorMsg(
+					ErrTypeMismatch,
+					fmt.Sprintf(`Cannot increment with non-numeric argument: {%s: %#v}`, incKey, incValue),
+				)
+			}
+
+			// $inc sets the field if it does not exist.
 			err := doc.SetByPath(path, incValue)
 			if err != nil {
 				return false, NewWriteErrorMsg(

